arrow-cast/src/cast/mod.rs

```diff
@@ -1413,10 +1437,34 @@ pub fn cast_with_options(
                 "Casting from {from_type:?} to {to_type:?} not supported",
             ))),
         },
-        (BinaryView, Binary) => cast_view_to_byte::<BinaryViewType, GenericBinaryType<i32>>(array),
-        (BinaryView, LargeBinary) => {
-            cast_view_to_byte::<BinaryViewType, GenericBinaryType<i64>>(array)
-        }
+        (BinaryView, _) => match to_type {
+            Binary => cast_view_to_byte::<BinaryViewType, GenericBinaryType<i32>>(array),
+            LargeBinary => cast_view_to_byte::<BinaryViewType, GenericBinaryType<i64>>(array),
+            Utf8 => {
+                let binary_arr =
+                    cast_view_to_byte::<BinaryViewType, GenericBinaryType<i32>>(array)?;
+                cast_binary_to_string::<i32>(&binary_arr, cast_options)
+            }
+            LargeUtf8 => {
+                let binary_arr =
+                    cast_view_to_byte::<BinaryViewType, GenericBinaryType<i64>>(array)?;
+                cast_binary_to_string::<i64>(&binary_arr, cast_options)
+            }
+            Utf8View => {
+                if let Some(arr) = array.as_any().downcast_ref::<BinaryViewArray>() {
+                    arr.clone()
+                        .to_string_view()
+                        .map(|x| Arc::new(x) as ArrayRef)
+                } else {
+                    Err(ArrowError::CastError(
+                        "Cannot cast BinaryView to StringView".to_string(),
+                    ))
+                }
+            }
```

Comment:
As above, I think this would be more consistent if it used `as_binary_view`:



```suggestion
            Utf8View => Ok(Arc::new(array.as_binary_view().clone().to_string_view()?) as ArrayRef),
```